### PR TITLE
283 : Add the multi process on the server

### PR DESCRIPTION
**Pull request by ZQUEMA**
## Purpose:
The purpose of this pull request is to add the possibility to create multiple rooms which run at the same time.

## Added features:
- Creation of an `r-type_room` executable which is used to start a room.
- The server can start and run multiple rooms at the same time thanks to multi-processing.
- The server and its rooms can communincate with each other. The protocole is here.

## Updated features:
- Renamed the library `RTypeLogicServer` to `RTypeLogicRoom`
- The `ArgumentHandler` class can now parse the input of an executable for the executable `r-type_room`

## Fixed features:

### CODE REVIEW

File: Room/main.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using namespace argument_handler;
+
+int main(int ac, char **av)
+{
+    ArgumentHandler argumentHandler = ArgumentHandler(ac, av);
+    ArgumentHandler::RoomInformation roomInformations = argumentHandler.extractRoomInformations();
+
+    serverData::Room room(
+        roomInformations.id, roomInformations.name, Client(roomInformations.address, roomInformations.port));
+    std::cout << "Room " << room.getRoomName() << " started" << std::endl;
```
Comment (LucasTesnier) [inline Room/main.cpp:21]: Print your login information in std::cerr

File: Server/RoomInstance.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/*
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** RoomInstance
+*/
+
+#include "Error/Error.hpp"
+#include "Server.hpp"
+
```
Comment (LucasTesnier) [inline Server/RoomInstance.cpp:10]: useless empty line

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some issue + windows working

**Review by TomDUVAL-MAHE — APPROVED** [review]

File: libs/ArgumentHandler/ArgumentHandler.hpp [hunk 3]
```diff
@@ -74,6 +89,10 @@ namespace argument_handler
         /// @return A struct containing all wanted informations.
         ClientInformation extractClientInformation(void);
 
+        /// @brief Extract all wanted value by the room from stored arguments.
```
[the remark this answers is not in the document]
Comment (ZQUEMA) [reply libs/ArgumentHandler/ArgumentHandler.hpp:92]: Wasn't my code but fair enough

File: Room/main.cpp (hunk 1 above)
Comment (ZQUEMA) [reply Room/main.cpp:21]: It shouldn't be here

**Review by LucasTesnier — DISMISSED** [review]

**Review by osvegn — DISMISSED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]
ntr good job !

**Pull request merged**